pkg/azuredisk/nodeserver_v2.go

```diff
@@ -519,3 +560,99 @@ func (d *DriverV2) ensureMountPoint(target string) (bool, error) {
 	klog.Warning("ensureMountPoint method is deprecated.")
 	return d.nodeProvisioner.EnsureMountPointReady(target)
 }
+
+func (d *DriverV2) recoverMount(diskURI string) {
+	// if mount recovery is necessary (if time.Now is past the set nextRetry), detach volume so that volume can be reattached via reconciliation
+	if d.shouldStartRecovery(diskURI) {
+		klog.Warningf("Starting mount recovery: detaching the volume (%s) from node (%s)...", diskURI, d.NodeID)
+		recoveryCtx := context.Background()
+
+		// When the device path cannot be found for the specified LUN, try to recover by detaching and re-attaching the disk.
+		// We only need to perform the detach. The CSI infrastructure will re-attach the disk when it notices a difference in
+		// world states (ListVolumes result doesn't match VolumeAttachment list).
+		detachErr := d.crdProvisioner.UnpublishVolume(recoveryCtx, diskURI, d.NodeID, nil)
+		if detachErr != nil {
+			klog.Errorf("failed to unpublishVolume volume (%s) from node (%s): %v", diskURI, d.NodeID, detachErr)
+			d.markDetachComplete(diskURI, false)
+		} else {
+			klog.Infof("Detached volume (%s) from node (%s) successfully...", diskURI, d.NodeID)
+			d.markDetachComplete(diskURI, true)
+		}
+	}
+}
+
+func (d *DriverV2) shouldStartRecovery(diskURI string) bool {
+	d.deviceChecker.lock.Lock()
+	defer d.deviceChecker.lock.Unlock()
+
+	recoveryInProcess, diskURIMatches := d.isRecoveryInProcess(diskURI, true)
+	if !recoveryInProcess {
+		d.deviceChecker.entry = newDeviceCheckerEntry(diskURI)
+	} else if !diskURIMatches {
+		// if there already is another recovery in process for different volume, no need to start recovery because recovery for that other volume will likely fix the problem for this volume
+		return false
+	}
+
+	// double check if diskURI match in case device checker was overwritten by a concurrent nodeStageCall
+	if d.deviceChecker.entry.diskURI != diskURI {
+		return false
+	}
+
+	if time.Now().Before(d.deviceChecker.entry.nextRetry) {
+		return false
+	}
+
+	d.deviceChecker.entry.detachInProcess = detachInProcess
+	return true
+}
+
+func (d *DriverV2) isRecoveryInProcess(diskURI string, lockAcquired bool) (recoveryInProcess, diskURIMatches bool) {
+	if !lockAcquired {
+		d.deviceChecker.lock.RLock()
+		defer d.deviceChecker.lock.RUnlock()
+	}
+	if d.deviceChecker.entry == nil {
+		recoveryInProcess = false
+		diskURIMatches = false
+	} else {
```

Comment:
`bool` values default to `false`, so you could eliminate the explicit assignments above.